ctapipe/core/tool.py

```diff
@@ -238,17 +240,22 @@ def run(self, argv=None):
             Provenance().finish_activity(activity_name=self.name, status="error")
             exit_status = 1  # any other error
         finally:
-            for activity in Provenance().finished_activities:
-                output_str = " ".join([x["url"] for x in activity.output])
-                self.log.info("Output: %s", output_str)
-
-            self.log.debug("PROVENANCE: '%s'", Provenance().as_json(indent=3))
-            self.provenance_log.parent.mkdir(parents=True, exist_ok=True)
-            with open(self.provenance_log, mode="a+") as provlog:
-                provlog.write(Provenance().as_json(indent=3))
+            print(self.argv)
+            if not {"-h", "--help", "--help-all"}.intersection(self.argv):
```

Comment:
I don't see a test to make sure the provlog is not written if only help is ran

Reply:
Good point.

Reply:
Added the test